### ISSUE TITLE
pytest-check fails with pytest's  --maxfail option

### ISSUE THREAD

**Issue by MichaelKorn** [okken/pytest-check#64]
pytest-check can not be used in combination with the `--maxfail` option because `@check_func` seems to have no effect and `AssertionError`s are not caught.

### minimal example
Use example from readme as `test_check.py`:
```python
def test_example(check):
    a = 1
    b = 2
    c = [2, 4, 6]
    check.greater(a, b)
    check.less_equal(b, a)
    check.is_in(a, c, "Is 1 in the list")
    check.is_not_in(b, c, "make sure 2 isn't in list")
```
start with `pytest.exe --maxfail=10  test_check.py`

### result
```
================================================ test session starts =================================================
platform win32 -- Python 3.7.9, pytest-6.2.4, py-1.10.0, pluggy-0.13.1
rootdir: XXXX
plugins: pytest_check-1.0.1, cov-2.12.1
collected 1 item

test_check.py F                                                                                                [100%]

====================================================== FAILURES ====================================================== 
____________________________________________________ test_example ____________________________________________________ 

check = <module 'pytest_check.check_methods' from 'XXXX\\lib\\site-packages\\pytest_check\\check_methods.py'>

    def test_example(check):
        a = 1
        b = 2
        c = [2, 4, 6]
>       check.greater(a, b)

test_check.py:5:
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _

a = 1, b = 2, msg = ''

    @check_func
    def greater(a, b, msg=""):
>       assert a > b, msg
E       AssertionError: 
E       assert 1 > 2

XXXX\lib\site-packages\pytest_check\check_methods.py:175: AssertionError
============================================== short test summary info ============================================== 
FAILED test_check.py::test_example - AssertionError:
================================================= 1 failed in 0.18s ================================================= 
```
Same results with linux (docker) with python 3.9.

**Comment by okken** [comment]
It currently works as:
- if maxfail == 1 (or -x), then only one failure is allowed. First failed check stops testing.
- if maxfail >= 2, then the maxfail count refers to the failing test functions, not the failing checks.

In other words, one test with 5 failed checks still counts as one failed test.
The exception to this rule is for -x or --maxfail=1, which then stops on the first failed check

I could see the argument in both directions of whether maxfail should count checks or functions.
Right now it most closely aligns with functions.
Changing to counting checks would be possible, but would change the behavior of the plugin.
At this point in it's adoption, I don't think I want to change the behavior.

However, documenting this behavior would be good.
So thank you for submitting this issue.

**Comment by okken** [comment]
closing this and opening #65

**Comment by MichaelKorn** [comment]
@okken Unfortunately, I cannot understand your answer.
But maybe I didn't make my ticket clear enough, I thought that would be reflected in the console output (With `--maxfail=10` the tests fails with the **first** `assert`).
To make my description a little clearer:
1. Without `--maxfail` the outputs are as expected and multiple asserts can fail within one test.
2. With `--maxfail` pytest-check doesn't seem to have any effect. Obviously, the PlugIn cannot be used in combination, the test fails with the first failed assert.

**Comment by okken** [comment]
Ok. Thanks for being persistent.
I can reproduce the error now.
